src/smt-inout/yosys_smt_parser.cc

```diff
@@ -0,0 +1,465 @@
+/// \file Parser for SMT-LIB2 generated by Yosys
+/// the use of it is
+///   1. translate data-type to bitvector declarations
+///     a. datatype : recursively defined...
+///     b. fix some 
+///   2. add a no-change transition function
+///   3. -- (currently no ) change variable name to their meaningful names
+
+#include <ilang/util/log.h>
+#include <ilang/util/str_util.h>
+#include <ilang/util/container_shortcut.h>
+#include <ilang/smt-inout/yosys_smt_parser.h>
+
+
+namespace ilang {
+namespace smt {
+
+//#define get_mod_name(s) ((s).substr(1,(s).length()-4))
+#define get_mod_name(s) (s)
+
+// |module1 inst| -> inst
+std::string get_mod_inst_name(const std::string & in) {
+  ILA_ASSERT(S_IN(' ',in));
+  ILA_ASSERT(in.back() == '|');
+  auto pos = in.find(' ')+1;
+  return in.substr(pos, in.length()-pos-1);
+} // get_mod_inst_name
+
+/// construct flatten_datatype (hierarchically)
+void YosysSmtParser::construct_flatten_dataype() {
+  for (auto && module_name : smt_ast.data_type_order) {
+    ILA_ASSERT(IN(module_name, smt_ast.datatypes));
+    auto & state_var_vec = smt_ast.datatypes[module_name];
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'state_var_vec' can be reduced.](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)